flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/StoreSink.java

```diff
@@ -85,10 +84,25 @@ public StoreSink(
     }
 
     private OneInputStreamOperator<RowData, Committable> createWriteOperator(
-            String initialCommitUser) {
+            CheckpointConfig checkpointConfig, String initialCommitUser) {
         if (compactionTask) {
             return new StoreCompactOperator(table, initialCommitUser, compactPartitionSpec);
         }
+
+        if (table.options().changelogProducer() == CoreOptions.ChangelogProducer.FULL_COMPACTION) {
+            long checkpointInterval = checkpointConfig.getCheckpointInterval();
+            long checkpointThreshold =
+                    table.options().changelogProducerFullCompactionTriggerInterval().toMillis()
+                            / checkpointInterval;
```

Comment:
I don't think it's good to bind to checkpoint interval. Checkpoints may be random:
- For example, some checkpoints will be skipped in case of failure.
- For example, there can be checkpoint triggered by source in the future.

It would be better to judge directly by time.